src/services/userService.js

```diff
@@ -0,0 +1,54 @@
+let Sequelize = require('sequelize');
+
+
+function UserService(model) {
+  this.model = model;
+}
+
+UserService.prototype.createUser = async function(user) {
+  let data = await this.model.create(user);
+  return data.dataValues;
+};
+
+UserService.prototype.getUser = async function(id) {
+  let data = await this.model.findOne({
+    where: { id }
+  });
+  return data;
+};
+
+UserService.prototype.deleteUser = async function(id) {
+  let user = await this.model.update(
+    { isDeleted: true },
+    {
+      where: { id },
+      returning: true,
+      plain: true
+    }
+  );
+  return user[1];
+};
+
+UserService.prototype.updateUser =  async function(id, payload) {
+  try {
+    let user = await this.model.update(payload, {
+      where: { id, isDeleted: false },
+      returning: true,
+      plain: true
+    });
+    return user ? user[1] : null;
+  } catch (exe) {
```

Comment:
Consider at least returning the error so that the error handling at the top level could do its job.

Reply:
resolved